### PR TITLE
[WIP] Refactor: Deprecate rz_str_dup and use rz_str_new everywhere

### LINKED ISSUE #1176: Resolve the clash between `strdup()`, `rz_str_dup()`, and `rz_str_new()`.
We have 3 similar functions:
- `strdup()` from libc
- `rz_str_dup()`
- `rz_str_new()`

Some places use `strdup()` directly, while some - either `rz_str_dup()` or `rz_str_new()`. We should decide which one to choose and stick to the one, I think.

```c
// Returns a new heap-allocated copy of str.
// XXX what's the diff with rz_str_dup ?
RZ_API char *rz_str_new(const char *str) {
    return str ? strdup(str) : NULL;
}

// TODO: deprecate?
RZ_API char *rz_str_dup(char *ptr, const char *string) {
    char *str = rz_str_new(string);
    free(ptr); // in case ptr == string
    return str;
}
```

See `librz/util/str.c`

### PR DESCRIPTION
**Pull request by rbrishabh**
Fixes #1176

Hi @XVilka ! I'll take up this issue!  You can give an initial review when you have some time!
I'll add more commits ASAP!

Thanks :)

### CODE REVIEW

File: librz/util/str.c [hunk 1]
```diff
@@ -699,7 +699,6 @@ RZ_API const char *rz_str_nstr(const char *s, const char *find, int slen) {
 }
 
 // Returns a new heap-allocated copy of str.
-// XXX what's the diff with rz_str_dup ?
 RZ_API char *rz_str_new(const char *str) {
```
Comment (wargio) [inline librz/util/str.c:702]: `rz_str_new` takes 1 argument but you just renamed all `rz_str_dup` to that but not changed the arguments.

**Review by wargio — CHANGES_REQUESTED** [review]

**Comment by rbrishabh** [comment]
yes, I see! adding more commits later today! thank you

File: librz/util/str.c (hunk 1 above)
Comment (ret2libc) [reply librz/util/str.c:702]: also, the old `rz_str_dup` was also freeing the `ptr` argument, which means that now all things will leak.

**Comment by wargio** [comment]
Hello @rbrishabh do you plan on working on this or i can close this PR?

**Comment by rbrishabh** [comment]
hi @wargio ! Due to some extra work, I'm not able to give enough time to this currently! I'm sorry!
I'll close this! :)

**Pull request closed without merging**